File: src/main/java/com/rainbow/task/entity/Task.java
package com.rainbow.task.entity;

import com.rainbow.common.entity.CoreEntity;
import com.rainbow.team.entity.Team;
import com.rainbow.user.entity.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "tasks")
public class Task extends CoreEntity {

    private String title;
    private Boolean useYn;
    private Boolean operateYn;
    private Boolean recordYn;
    private Boolean deployYn;
    private Integer rpaScriptId;
    @Column(name = "task_order")
    private Integer order;

    /* 관계형 설정 */
    @ManyToOne
    @JoinColumn(name = "team_id")
    private Team team;

    @ManyToOne
    @JoinColumn(name = "work_id")
    private Work work;

    @ManyToOne
    @JoinColumn(name = "created_by_id")
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "updated_by_id")
    private User updatedBy;
}

File: task/entity/task.py
from sqlalchemy import Column, String, Integer

from common.database.db import Base
from common.entity.core_entity import CoreEntity


class Task(Base, CoreEntity):
    __tablename__ = "tasks"
    title = Column(String, unique=True, nullable=False)
    description = Column(String)
    keyword = Column(String)
    count = Column(Integer)

    # 아래 부터는 스케쥴 관련 필드 (테스트 후 Table 분리 예정)
    hour = Column(Integer)
    minute = Column(Integer)
    # interval 이 없으면 해당 시간에 1회만 실행
    # interval 이 있으면 해당 시간 부터 지속 적으로 실행
    interval = Column(Integer, nullable=True)
